Testscripts/Linux/validate-da.sh

```diff
@@ -0,0 +1,163 @@
+	install_log_version=$(awk "NR==3" /var/opt/microsoft/dependency-agent/log/install.log | grep 'Dependency Agent version.revision'|cut -f2 -d ":")
+	if [ "$bin_version" -ne "$install_log_version" ]; then
+	        LogErr "Version mismatch between bin version and install log version"
+	        exit 1
+	else
+	        LogMsg "Version matches between bin version and install log version"
+	fi
+
+	LogMsg "Install tests passed successfully"
+}
+
+function verify_uninstall_da() {
+
+	bash /opt/microsoft/dependency-agent/uninstall
+	
+	ret=$?
+	if [ $ret -ne 0 ]; then
+	        LogErr "Uninstall failed with exit code ${ret}"
+	        exit 1
+	else
+	        LogMsg "Uninstalled with exit code 0"
+	fi
+
+	[ -d "/opt/microsoft/dependency-agent" ] && LogErr "Directory /opt/microsoft/dependency-agent exists." && exit 1
+	[ -d "/var/opt/microsoft/dependency-agent" ] && LogErr "Directory /var/opt/microsoft/dependency-agent exists." && exit 1
+	[ -d "/etc/opt/microsoft/dependency-agent" ] && LogErr "Directory /etc/opt/microsoft/dependency-agent exists." && exit 1
+	[ -d "/lib/microsoft-dependency-agent" ] && LogErr "Directory /lib/microsoft-dependency-agent exists." && exit 1
+	[ -d "/etc/init.d/microsoft-dependency-agent" ] && LogErr "Directory /etc/init.d/microsoft-dependency-agent exists." && exit 1
+
+	LogMsg "Uninstall tests passed successfully"
+}
+
+function setup_network_trace() {
+    LogMsg "Setup network trace"
+	watch -n 5 curl https://microsoft.com &>/dev/null &
+	watch_pid=$!
+}
+
+function enable_disable_da(){
+	verify_install_da
+	
+	sleep 30
+	[ ! -f "/var/opt/microsoft/dependency-agent/log/service.log" ] && LogErr "/var/opt/microsoft/dependency-agent/log/service.log does not exist." && exit 1
+
+	service_log_time=$(date -r /var/opt/microsoft/dependency-agent/log/service.log +%s)
+	current_time=$(date +%s)
+	time_elapsed=$(($current_time - $service_log_time))
+	sleep $((120-$time_elapsed))
+
+	[ ! -f "/var/opt/microsoft/dependency-agent/log/service.log.1" ] && LogErr "/var/opt/microsoft/dependency-agent/log/service.log.1 does not exist." && exit 1
+	[ -f "/var/opt/microsoft/dependency-agent/log/service.log.2" ] && LogErr "/var/opt/microsoft/dependency-agent/log/service.log.2 exist." && exit 1
+	[ ! -c "/dev/msda" ] && LogErr "/dev/msda does not exist." && exit 1
+
+	! grep -iq "driver setup status=0" /var/opt/microsoft/dependency-agent/log/service.log.1 && LogErr "driver setup status=0 not found" && exit 1
+	! grep -iq "starting the dependency agent" /var/opt/microsoft/dependency-agent/log/service.log && LogErr "starting the dependency agent not found" && exit 1
+	
+	sleep 60
+	[ ! -f "/var/opt/microsoft/dependency-agent/log/MicrosoftDependencyAgent.log" ] && LogErr "/var/opt/microsoft/dependency-agent/log/MicrosoftDependencyAgent.log does not exist." && exit 1
+	[ ! -f "/etc/opt/microsoft/dependency-agent/config/DA_PID" ] && LogErr "/etc/opt/microsoft/dependency-agent/config/DA_PID does not exist." && exit 1
```

Comment:
Many of the pathnames, such as /etc/opt/microsoft/dependency-agent/config/DA_PID, are used many times in the script. Please consider refactoring them into symbolic constants (i.g. `readonly da_pid_file="/etc/opt/microsoft/dependency-agent/config/DA_PID" ). This will help readability and reduces the changes of various types of bugs.

Reply:
Made this into a small function to verify DA_PID file